hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -74,6 +96,130 @@ private boolean hasCryptoAdminKey(final ContractUpdateTransactionBody op) {
+        }
+        if (op.hasAutoRenewAccountId()) {
+            builder.autoRenewAccountId(op.autoRenewAccountId());
+        }
+        if (op.hasMaxAutomaticTokenAssociations()) {
+            final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+            final var entitiesConfig = context.configuration().getConfigData(EntitiesConfig.class);
+            final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+
+            validateFalse(
+                    op.maxAutomaticTokenAssociations() > ledgerConfig.maxAutoAssociations(),
+                    REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT);
+
+            final long newMax = op.maxAutomaticTokenAssociations();
+            validateFalse(newMax < contract.maxAutoAssociations(), EXISTING_AUTOMATIC_ASSOCIATIONS_EXCEED_GIVEN_LIMIT);
+            validateFalse(
+                    entitiesConfig.limitTokenAssociations() && newMax > tokensConfig.maxPerAccount(),
+                    REQUESTED_NUM_AUTOMATIC_ASSOCIATIONS_EXCEEDS_ASSOCIATION_LIMIT);
+
+            builder.maxAutoAssociations(op.maxAutomaticTokenAssociations());
+        }
+        return builder.build();
+    }
+
+    private void validateSemantics(Account contract, HandleContext context, ContractUpdateTransactionBody op) {
+        validateTrue(contract != null, INVALID_CONTRACT_ID);
+
+        if (op.hasAdminKey()) {
+            boolean keyNotSentinel = !EMPTY_KEY_LIST.equals(op.adminKey());
+            boolean keyIsUnset = op.adminKey().key().kind() == KeyOneOfType.UNSET;
+            boolean keyIsNotValid = !KeyUtils.isValid(op.adminKey());
+            validateFalse(keyNotSentinel && (keyIsUnset || keyIsNotValid), INVALID_ADMIN_KEY);
+        }
+
+        if (op.hasExpirationTime()) {
+            try {
+                context.attributeValidator().validateExpiry(op.expirationTime().seconds());
+            } catch (HandleException e) {
+                validateFalse(contract.expiredAndPendingRemoval(), CONTRACT_EXPIRED_AND_PENDING_REMOVAL);
+                throw e;
+            }
+        }
+
+        validateFalse(!onlyAffectsExpiry(op) && !isMutable(contract), MODIFYING_IMMUTABLE_CONTRACT);
+        validateFalse(reducesExpiry(op, contract.expirationSecond()), EXPIRATION_REDUCTION_NOT_ALLOWED);
+
+        // validate expiry metadata
+        final var currentMetadata =
+                new ExpiryMeta(contract.expirationSecond(), contract.autoRenewSeconds(), contract.autoRenewAccountId());
+        final var updateMeta = new ExpiryMeta(
+                op.hasExpirationTime() ? op.expirationTime().seconds() : NA,
+                op.hasAutoRenewPeriod() ? op.autoRenewPeriod().seconds() : NA,
+                null);
+        context.expiryValidator().resolveUpdateAttempt(currentMetadata, updateMeta);
+    }
+
+    boolean onlyAffectsExpiry(ContractUpdateTransactionBody op) {
+        return !(op.hasProxyAccountID()
```

Comment:
:warning: This `||` conjunction is missing `op.hasMaxAutoAssociations()`...I see it's **also** missing in mono-service 😞---though it shouldn't really matter in either place given we'll always be failing with `NOT_SUPPORTED` if it's set.